WordPress/src/main/java/org/wordpress/android/ui/utils/UiHelpers.kt

```diff
@@ -16,4 +20,36 @@ class UiHelpers @Inject constructor() {
     fun updateVisibility(view: View, visible: Boolean) {
         view.visibility = if (visible) View.VISIBLE else View.GONE
     }
+
+    fun setTextOrHide(view: TextView, uiString: UiString?) {
```

Comment:
Does the `UiHelpers` class mean that we're moving away from Kotlin extensions? Or this is just a special case?

Reply:
I'm personally not a fan of **public/global** extension functions. I feel like they are really easy to misuse. Having said that I'm not strictly against them so if you both feel an extension function would be a better solution, I'll refactor it;).

Reply:
I feel like I should have elaborated more:

- It's hard to know where is the extension function defined.
- If you don't know the api well enough you might not even know it's an extension function and not a part of the api. It might make finding a bug more difficult because in most cases I assume the api is tested and works and I look for a bug in the api itself in the last step.
- Extension functions can't be mocked. This is going back to my previous comment - I don't like static access.
- This doesn't concern me too much but naming collision can be an issue as well (applies too private extension functions as well)

Reply:
> Having said that I'm not strictly against them so if you both feel an extension function would be a better solution, I'll refactor it;).

This is fine. I'm mainly commenting here to understand the general consensus. 🙂So thank you for answering my questions. 

> It's hard to know where is the extension function defined.

This usually happens when there are too many. Though I find extension functions to be more discoverable. For example, as I type something like "`stringVariable.`", I get autocompletion for all available string extensions. But I can see how this discussion will turn into a "to each his own" scenario. 😁

Reply:
@malinajirka's comment sums up really well why we didn't go with the extensions. I agree with the discoverability bit, but I am a defensive programmer, so I tend to pick the solution that's less likely to create headache for me if two options are somewhat equal. Having said that if there is a strong case to change it, I am not against it either.

> But I can see how this discussion will turn into a "to each his own" scenario.

Pretty much this 😁

Reply:
> Pretty much this 😁

Yeah, as I said, I'm okay with this as is. I need some more time to get the whole picture of the source to build really strong opinions. I just wanted to share my thoughts. 

Thank you for responding!